diff -urNp gcc-linaro-4.6-2013.05.orig/gcc/config/arm/linux-eabi.h gcc-linaro-4.6-2013.05/gcc/config/arm/linux-eabi.h
--- gcc-linaro-4.6-2013.05.orig/gcc/config/arm/linux-eabi.h	2014-02-07 19:05:59.000000000 +0900
+++ gcc-linaro-4.6-2013.05/gcc/config/arm/linux-eabi.h	2014-05-09 20:07:09.042410813 +0900
@@ -60,23 +60,10 @@
 #undef  SUBTARGET_EXTRA_LINK_SPEC
 #define SUBTARGET_EXTRA_LINK_SPEC " -m " TARGET_LINKER_EMULATION
 
-/* GNU/Linux on ARM currently supports three dynamic linkers:
-   - ld-linux.so.2 - for the legacy ABI
-   - ld-linux.so.3 - for the EABI-derived soft-float ABI
-   - ld-linux-armhf.so.3 - for the EABI-derived hard-float ABI.
-   All the dynamic linkers live in /lib.
-   We default to soft-float, but this can be overridden by changing both
-   GLIBC_DYNAMIC_LINKER_DEFAULT and TARGET_DEFAULT_FLOAT_ABI.  */
-
+/* Use ld-linux.so.3 so that it will be possible to run "classic"
+   GNU/Linux binaries on an EABI system.  */
 #undef  GLIBC_DYNAMIC_LINKER
-#define GLIBC_DYNAMIC_LINKER_SOFT_FLOAT "/lib/ld-linux.so.3"
-#define GLIBC_DYNAMIC_LINKER_HARD_FLOAT "/lib/ld-linux-armhf.so.3"
-#define GLIBC_DYNAMIC_LINKER_DEFAULT GLIBC_DYNAMIC_LINKER_SOFT_FLOAT
-
-#define GLIBC_DYNAMIC_LINKER \
-   "%{mfloat-abi=hard:" GLIBC_DYNAMIC_LINKER_HARD_FLOAT "} \
-    %{mfloat-abi=soft*:" GLIBC_DYNAMIC_LINKER_SOFT_FLOAT "} \
-    %{!mfloat-abi=*:" GLIBC_DYNAMIC_LINKER_DEFAULT "}"
+#define GLIBC_DYNAMIC_LINKER "/lib/ld-linux.so.3"
 
 /* At this point, bpabi.h will have clobbered LINK_SPEC.  We want to
    use the GNU/Linux version, not the generic BPABI version.  */
